File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        // int[] freq = new int[501];
        // int max = 0, result = -1;
        // for (int i = 0; i < arr.length; i ++) {
        //     freq[i]++;
        //     max = Math.max(arr[i], max);
        // }

        // for (int i = max; i > 0; i--) {
        //     if (i == freq[i]) {
        //         result = i;
        //         break;
        //     }
        // }

        // return result;
        int result = -1;
        
        for (int i = arr.length - 1; i >= 0; i--) {
            int freq = 0, val = i + 1;
            for (int num : arr) {
                if (num == val) freq++;
                if (freq > val) break;                
            }

            if (freq == val) {
                result = val;
                break;
            }
        }

        return result;
    }
}

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        freq = [0] * 501
        max_val = 0
        result = -1

        for num in arr:
            freq[num] += 1
            max_val = max(max_val, num)

        for i in range(max_val, 0, -1):
            if freq[i] == i:
                result = i
                break

        return result